packaging/rpm/amanda.spec.src

```diff
@@ -26,135 +26,30 @@
 
 # Pkg-config sometimes needs its own path set, and we need to allow users to
 # override our guess during detection.  This macro takes care of that.
-# If no --define PKG_CONFIG_PATH was passed and env var $PKG_CONFIG_PATH is 
+# If no --define PKG_CONFIG_PATH was passed and env var $PKG_CONFIG_PATH is
 # set then use the env var.
 %{!?PKG_CONFIG_PATH: %{expand:%(echo ${PKG_CONFIG_PATH:+"%%define PKG_CONFIG_PATH $PKG_CONFIG_PATH"})}}
 
 # LDFLAGS: newer gnu linkers can use "-Wl,--as-needed", while older ones,
 # like rhel3 and sles9 can't.
 %define enable_as_needed --enable-as-needed
 
+%define curl libcurl
+
+%define dist       %%PKG_DIST%%
+%define distver    %%PKG_DISTVER%%
+%define disttag    %(A="%%PKG_SUFFIX%%"; A="${A#.}"; echo ${A%%%%.*})
+
 # Define which Distribution we are building:
 # Try to detect the distribution we are building:
 %if %{_vendor} == amazon
     %define dist amazon
-    %define disttag amzn
     %define distver 1
-    %define curl libcurl
-%endif
```

Comment:
It is not easy nor preferable to maintain an if-tree based off the _vendor and scrouging around files in RPM spec stuff.